client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,146 @@
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constrainVideoDimensions( shortcodeWidthAttribute, shortcodeHeightAttribute ) {
+		const defaultWidth = 640;
+		const defaultAspectRatio = 16 / 9;
+		let { width, height } = this.props.video
+			? this.props.video
+			: {
+				width: defaultWidth,
+				height: defaultWidth / defaultAspectRatio
+			};
+		const aspectRatio = width / height;
+
+		if ( shortcodeWidthAttribute && ! shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeWidthAttribute / aspectRatio;
+		} else if ( ! shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeHeightAttribute * aspectRatio;
+			height = shortcodeHeightAttribute;
+		} else if ( shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			const definedAspectRatio = shortcodeWidthAttribute / shortcodeHeightAttribute;
+			if ( definedAspectRatio > aspectRatio ) {
+				width = shortcodeHeightAttribute * aspectRatio;
+				height = shortcodeHeightAttribute;
+			} else {
+				width = shortcodeWidthAttribute;
+				height = shortcodeWidthAttribute / aspectRatio;
+			}
+		}
+
+		return { width, height };
+	}
+
+	getVideoAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+
+		const videoDimensions = this.constrainVideoDimensions(
+			parseInt( namedAttrs.w, 10 ),
+			parseInt( namedAttrs.h, 10 )
+		);
+
+		const shortcodeAttributes = {
+			guid: shortcode.attrs.numeric[0],
+			w: videoDimensions.width,
+			h: videoDimensions.height,
+			autoplay: namedAttrs.autoplay === 'true',
+			hd: namedAttrs.hd === 'true',
+			loop: namedAttrs.loop === 'true',
+			at: parseInt( namedAttrs.at, 10 ) || 0,
+			defaultLangCode: namedAttrs.defaultlangcode
+		};
+
+		return Object.assign(
+			{},
+			pick( shortcodeAttributes, ['guid', 'w', 'h'] ),
+			{ embedUrl: this.getEmbedUrl( shortcodeAttributes ) }
+		);
+	}
+
+	getEmbedUrl( shortcodeAttributes ) {
+		const defaultAttributeValues = { hd: false, at: 0, defaultLangCode: undefined };
+		const attributesWithNonDefaultValues = omitBy(
+			shortcodeAttributes,
+			( value, key ) => defaultAttributeValues[key] === value );
+		const queryStringAttributes = ['autoplay', 'hd', 'loop', 'at', 'defaultLangCode'];
+		const queryString = QueryString.stringify( pick( attributesWithNonDefaultValues, queryStringAttributes ) );
+
+		return `https://videopress.com/embed/${ shortcodeAttributes.guid }?${ queryString }`;
+	}
+
+	onLoad() {
+		const doc = this.props.editor.iframeElement.contentDocument;
+		const script = doc.createElement( 'script' );
+		script.src = 'https://videopress.com/videopress-iframe.js';
```

Comment:
The included script takes care of finding and resizing the iframe we render, and does so by maintaining the aspect ratio. At the same time it also resizes the divs embedded in the iframe for the video, the player and whatnot, and also updates the width/height attribute of the video tag (it's quite deeply nested).

Without the resize script the iframe width is computed dynamically, but we could have to compute and update its height, but that's not enough as I outlined above: we should update dimensions of elements *within* the VideoPress iframe. I would keep the existing script so that we do not add knowledge to Calypso about how VideoPress videos are rendered: we provide an iframe, the rest is taken care if by VideoPress.